pkg/api/plugin.go

```diff
@@ -122,14 +113,14 @@ type Plugin interface {
 	// (for update, upgrade, etc.)
 	// externalOnly indicates that fields set by the RP (FQDN and routerProfile.FQDN)
 	// should be excluded.
-	Validate(ctx context.Context, new, old *OpenShiftManagedCluster, externalOnly bool) []error
+	Validate(ctx context.Context, new, old *OpenShiftManagedCluster, pc PluginConfig, externalOnly bool) []error
```

Comment:
use config from validator, and not pass

Reply:
ValidateTemplate() as separate function

Reply:
Include yaml file and read the file for pluginConfig and use it as tempalte/base

Reply:
Why does PluginConfig have to be passed as an argument in the plugin functions vs being an internal implementation detail of the interface implementation?

Reply:
I see this is related to https://github.com/openshift/openshift-azure/issues/835 - nevermind